### PR TITLE
webdav: add support for WebDav digest auth - fixes #2110

### LINKED ISSUE #2110: Support for digest authenticaton
Hi:

rclone doesn’t support digest authenticaton yet.
You can see this [link](https://forum.rclone.org/t/head-not-found-with-webdav/4935) in detail.
How about support digest authenticaton?
English is not my native language; please excuse typing errors.

### PR DESCRIPTION
**Pull request by socram8888**
#### What is the purpose of this change?

This PR adds support for Digest authentication for the WebDav backend.

Arguably, this implementation is suboptimal, in the sense that Basic authentication is always sent on first request, which nulls out any security benefits of using digest authentication, but it's enough to get talking to servers which only accept Digest authentication (such as my [Sandisk Wireless drive](https://kb.sandisk.com/app/answers/detail/a_id/17367/related/1)).

#### Was the change discussed in an issue or in the forum before?

Previously discussed in #2110

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/webdav/webdav.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ import (
 	"github.com/rclone/rclone/lib/pacer"
 	"github.com/rclone/rclone/lib/rest"
 
+	"github.com/icholy/digest"
```
Comment (albertony) [inline backend/webdav/webdav.go:41]: Linting error here, from the goimports tool. Think it needs to be moved up to just before the first "github.com/rclone" (sorted order).

Comment (socram8888) [reply backend/webdav/webdav.go]: I've just moved it up to where it belongs.

Go's really not my thing, so I thought this belonged below, along with ntlmss (another external import)

**Comment by Drakonas** [comment]
Is there any progress on this? It's been sitting for a while now.

**Comment by xiconet** [comment]
Me too

**Comment by ncw** [comment]
This needs re-doing for the chi router which we are now using. I didn't find a pre-build one so you'd want to make one using this https://github.com/go-chi/chi/blob/master/middleware/basic_auth.go as a starting point.

**Pull request closed without merging**